README.md

```diff
@@ -18,7 +19,7 @@ on:
     branches:
       - main
 jobs:
-  build:
+  deploy:
```

Comment:
nit: build and deploy?

Reply:
`build-and-deploy` looks... clunky? This doesn't show up as the name, so just describing the job's ultimate value (deploying the app) seems fine? Consistent with our [other docs](https://aws.amazon.com/blogs/apn/using-gitlab-ci-cd-pipeline-to-deploy-aws-sam-applications/) and some others that I can't find right now.